File: Two Sums/TwoSum.cpp
#include<stdio.h>
#include<vector>
#include<iostream>
using namespace std;

vector<int> TwoSum(vector<int> num, int target){
    //  for (int i = 0; i < num.size(); i++) {
    //     // Printing the element at
    //     // index 'i' of vector
    //     cout << num[i];
    // }

    vector<int> ret;
    int size = num.size();
    for(int i = 0; i < size -1; i++){
        for(int j = i+1; j < size; j++){
            if(num[i] + num[j] == target){
                ret.push_back(i);
                ret.push_back(j);
                return ret;

            }
        }
    }

}
int main(){
    vector<int> num{2,7,11,15}; 
    int target = 9;

    vector<int> numRet = TwoSum(num, target);
    for(int i = 0; i <  numRet.size(); i++){
        cout << numRet[i];
    }


return 0;
}

File: DSA/Two Sums/TwoSum.java
public class TwoSum{
    
    
    public static int[] TwoSumBruteForce(int[] num, int target){
        for(int i = 0; i < num.length; i++ ){
            for(int j =0; j < num.length; j++){
                int complement = target - num[i];

                if(num[j] ==  complement){
                    return new int []{i, j};
                }
            }
        }
        return new int[]{};
    }

    
    public static void main(String[] args){
        int num[] = {2,7,11,15};
        int target = 9;
        int[] ret = TwoSumBruteForce(num, target);
        for(int i = 0; i<ret.length; i++){
            System.out.println(ret[i]);
        }
    }
}